MapboxCoreNavigation/RouteController.swift

```diff
@@ -675,6 +670,27 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    /**
+     Monitors the user's course to see if it is consistantly moving away from what we expect the course to be at a given point.
+     */
+    func isCourseIsOnRoute(_ location: CLLocation) -> Bool {
+        let nearByCoordinates = routeProgress.currentLegProgress.nearbyCoordinates
+        
+        guard let calculatedCourseForLocationOnStep = interpolatedCourse(from: location, along: nearByCoordinates) else { return true }
+        
+        let maxUpdatesAwayFromRouteGivenAccuracy = Int(location.horizontalAccuracy / 4)
+        
+        if movmentsAwayFromRoute >= max(3, maxUpdatesAwayFromRouteGivenAccuracy)  {
+            return false
+        } else if shouldSnap(location, toRouteWith: calculatedCourseForLocationOnStep) {
```

Comment:
For potential debug purpose, I would rather assign this computed condition into a variable. Just a thought...